libs/dmlf/include/dmlf/colearn/muddle_learner_networker_impl.hpp

```diff
@@ -146,7 +147,27 @@ class MuddleLearnerNetworkerImpl : public AbstractMessageController
   Address     GetAddress() const;
   std::string GetAddressAsString() const;
 
+  void SetShuffleAlgorithm(const std::shared_ptr<ShuffleAlgorithmInterface> &alg);
```

Comment:
```suggestion
  void SetShuffleAlgorithm(std::shared_ptr<ShuffleAlgorithmInterface> const &alg);
```

Maybe it is better to a copy instead of a const& also.

Reply:
I think the analyser wants it passed-by ref and std::moved.